mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxUpdateParentAsLookupListener.java

```diff
@@ -0,0 +1,103 @@
+import org.apache.james.mailbox.store.event.EventFactory.MailboxLookupRightUpdate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MailboxUpdateParentAsLookupListener implements MailboxListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MailboxUpdateParentAsLookupListener.class);
+
+    private final MailboxSessionMapperFactory mailboxSessionMapperFactory;
+    private final StoreRightManager storeRightManager;
+
+    @Inject
+    public MailboxUpdateParentAsLookupListener(MailboxSessionMapperFactory mailboxSessionMapperFactory, StoreRightManager storeRightManager) {
+        this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
+        this.storeRightManager = storeRightManager;
+    }
+
+    @Override
+    public ListenerType getType() {
+        return ListenerType.ONCE;
+    }
+
+    @Override
+    public ExecutionMode getExecutionMode() {
+        return ExecutionMode.ASYNCHRONOUS;
+    }
+
+    @Override
+    public void event(Event event) {
+        if (event instanceof EventFactory.MailboxLookupRightUpdate) {
+            MailboxLookupRightUpdate aclUpdateEvent = (MailboxLookupRightUpdate) event;
+
+            MailboxSession mailboxSession = aclUpdateEvent.getSession();
+            MailboxPath mailboxPath = aclUpdateEvent.getMailboxPath();
+            List<MailboxPath> mailboxPaths = mailboxPath.getHierarchyLevels(mailboxSession.getPathDelimiter());
+            mailboxPaths.stream()
+                .filter(hierarchyMailboxPath -> !hierarchyMailboxPath.equals(mailboxPath))
+                .forEach(parentMailboxPath -> {
+                    updateLookupRight(parentMailboxPath, mailboxSession, aclUpdateEvent.getAclDiff());
+                });
+        }
+    }
+
+    private void updateLookupRight(MailboxPath mailboxPath, MailboxSession session, ACLDiff aclDiff) {
+        aclDiff.addedEntries()
+            .forEach(entry -> {
+                updateRight(mailboxPath, session, entry);
+        });
+        aclDiff.changedEntries()
+            .forEach(entry -> {
+                updateRight(mailboxPath, session, entry);
+        });
+    }
+
+    private void updateRight(MailboxPath mailboxPath, MailboxSession session, MailboxACL.Entry entry) {
+        MailboxACL.ACLCommand mailboxACLCommand = null;
+        try {
+            mailboxACLCommand = MailboxACL.command()
+                .rights(MailboxACL.Right.Lookup)
+                .key(entry.getKey())
+                .asAddition();
+            storeRightManager.applyRightsCommand(mailboxPath, mailboxACLCommand, session);
+        } catch (UnsupportedRightException e) {
+            LOGGER.info(String.format("User '%s' would like to share mailbox but does not support the right",
+                session.getUser().getUserName()), e);
+        } catch (MailboxException e) {
+            LOGGER.info(String.format("User '%s' would like to share mailbox but mailbox does not find",
```

Comment:
The exceptions says "something BAD happend on th mailbox level"

Which is different from mailbox not found.

 - > Catch mailboxNotFound in a separate block if you wish to handle it separatly
 - > MilboxException should be handled with Error level!